spec/models/materialized_patient_summary_spec.rb

```diff
@@ -1,279 +1,265 @@
 require "rails_helper"
 
 describe MaterializedPatientSummary, type: :model do
-  include QuarterHelper
+  subject(:patient_summary) { MaterializedPatientSummary.find(patient.id) }
+
+  let(:old_date) { DateTime.new(2019, 1, 1) }
+  let(:new_date) { DateTime.new(2019, 5, 1) }
+  let(:old_quarter) { "2019 Q1" }
+  let(:new_quarter) { "2019 Q2" }
+  let(:user) { create(:admin, :power_user) }
+  let(:facility) { create(:facility) }
+
+  # We create a bunch of related objects here to ensure the Materialized view properly rolls things up.
+  # We _do not_ split things out into individual let blocks, because it makes detangling dependancies too difficult.
+  let(:patient) do
+    patient = create(:patient, registration_facility: facility, recorded_at: old_date, registration_user: user)
+    _old_bp = create(:blood_pressure, patient: patient, facility: facility, recorded_at: old_date, user: user)
+    _old_phone = create(:patient_phone_number, patient: patient, device_created_at: old_date)
+    _new_phone = create(:patient_phone_number, patient: patient, device_created_at: new_date)
+    _old_passport = create(:patient_business_identifier, patient: patient, device_created_at: old_date)
+    @new_bp = create(:blood_pressure, patient: patient, facility: facility, recorded_at: new_date, user: user, systolic: 110, diastolic: 70)
+    @blood_sugar = create(:blood_sugar, patient: patient, facility: facility, recorded_at: new_date, user: user,
+                                        blood_sugar_type: "random", blood_sugar_value: 100)
+    patient
+  end
+
+  attr_reader :new_bp
+  attr_reader :blood_sugar
```

Comment:
I guess this is a natural outcome once we use instance vars to avoid mutating them. This reads like we're adding behaviour to a test though. I haven't seen this pattern before but it doesn't feel idiomatic. wdyt?

Reply:
I do understand Rob's desire to avoid `let!` because they do tend to bloat tests over time as different developers add to large test files, but I also have always disliked using instance variables in rspec. You can find articles with good reasons to avoid it. In this particular case, I can see why Rob is doing it. The previous implementation had several lets that were not directly used, which does make it more difficult to understand what's really going on and what's important. So I can support creating those within the patient definition. But this does feel very non-idiomatic. Personally, I would prefer to keep `new_bp` and `blood_sugar` in lets and add them to the patient in a before block.